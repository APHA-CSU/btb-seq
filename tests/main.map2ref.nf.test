nextflow_process {

    name "Test Process map2Ref"
    script "main.nf"
    process "map2Ref"

    test("Should maps reads to reference genome") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = ["tinyreads", file("${projectDir}/tests/data/tinyreads/dedup_1.fastq"), file("${projectDir}/tests/data/tinyreads/dedup_2.fastq"), file("${projectDir}/references/Mycbovis-2122-97_LT708304.fas")]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
