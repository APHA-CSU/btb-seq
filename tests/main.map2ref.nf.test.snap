{
    "Should maps reads to reference genome": {
        "content": [
            {
                "0": [
                    [
                        "tinyreads",
                        "tinyreads.bam:md5,48100a95c6836316872ed70cb50a39e2"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-27T21:50:19.803221393"
    }
}